Tools/ardupilotwaf/chibios.py

```diff
@@ -59,6 +60,20 @@ def run(self):
         # Refer Tools/scripts/macos_remote_upload.sh for details
         if 'AP_OVERRIDE_UPLOAD_CMD' in os.environ:
             cmd = "{} '{}'".format(os.environ['AP_OVERRIDE_UPLOAD_CMD'], src.abspath())
+        elif "microsoft-standard-WSL2" in platform.release():
+            # As of Feb 2022 WSL2 does not support native USB support. The workaround from Microsoft using 'usbipd' does not work due to the following workflow:
+            # 1) connect USB device to Windows compuiter running WSL2
+            # 2) device boots into app
+            # 3) use 'usbipd' from Windows Cmd/PowerShell to determine busid, this is very hard to automate on Windows
+            # 4) use 'usbipd' from Windows Cmd/PowerShell to attach, this is very hard to automate on Windows
+            # -- device is now viewable via 'lsusb' but you need sudo to read from it.
+            # either run 'chmod666 /dev/ttyACM*' or use udev to automate chmod on device connect
+            # 5) uploader.py detects device, sends reboot command which disconnects the USB port and reboots into bootloader (different USB device)
+            # 6) manually repeat steps 3 & 4
+            # 7) doing steps 3 and 4 will most likely take several seconds and in many cases the bootlaoder has moved on into the app
+            #
+            # Solution: simply call "python.exe" instead of 'python' which magically calls it from the windows system using the same absolute path back into the WSL's user's directory
+            cmd = "{} '{}/uploader.py' '{}'".format('python.exe', upload_tools, src.abspath())
```

Comment:
Our environment installer doesn't install this and it is not installed by default - you should check that it exists first and otherwise fallback to the regular invocation

Reply:
an error indication if not exists would be good.

Reply:
I've reworked this to spew out some self-documenting instructions